src/diem/offchain/command.py

```diff
@@ -25,10 +27,17 @@ def reference_id(self) -> str:
         ...
 
     @abstractmethod
-    def new_command(self) -> "Command":
-        """sub-class should define **kwargs for creating new command from current instance"""
+    def validate(self, prior: typing.Optional["Command"]) -> None:
```

Comment:
same here, command validation logic with prior command should be live here.

Reply:
Could you elaborate? Not sure I understand the remark.

Reply:
see comment below for the funds_pull_pre_approval_command, this comment was put at wrong place.